src/App.jsx

```diff
@@ -14,33 +15,91 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const ALPHABET = 'alphabet';
+const LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function prepareGoods(sort, reverse) {
+  const goods = [...goodsFromServer];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  goods.sort((good1, good2) => {
+    switch (sort) {
+      case ALPHABET:
+        return good1.localeCompare(good2);
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+      case LENGTH:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (reverse) {
+    goods.reverse();
+  }
+
+  return goods;
+}
+
+export const App = () => {
+  const [sort, setSort] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
+  const newGoods = prepareGoods(sort, isReverse);
+
+  const reset = () => {
+    setSort('');
+    setIsReverse(false);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${sort !== ALPHABET ? 'is-light' : ''}`}
+          onClick={() => {
+            setSort(ALPHABET);
+          }}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={`button is-success ${sort !== LENGTH ? 'is-light' : ''}`}
+          onClick={() => {
+            setSort(LENGTH);
+          }}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={`button is-warning ${!isReverse ? 'is-light' : ''}`}
+          onClick={() => {
+            setIsReverse(!isReverse);
+          }}
+        >
+          Reverse
+        </button>
+        {(isReverse || sort) && (
```

Comment:
Create a variable for `isReverse || sort` so you don't need to combine different logical operators and rename them, sort isn't goo name